api/src/test/java/org/pmiops/workbench/api/ProfileControllerTest.java

```diff
@@ -582,31 +581,29 @@ private Profile createUser() throws Exception {
 
   private void assertProfile(Profile profile, String primaryEmail, String contactEmail,
       String familyName, String givenName, DataAccessLevel dataAccessLevel,
-      Timestamp firstSignInTime, String freeTierBillingProject, boolean enabledInFirecloud,
-      Boolean contactEmailFailure) {
+      Timestamp firstSignInTime, boolean enabledInFirecloud, Boolean contactEmailFailure) {
     assertThat(profile).isNotNull();
     assertThat(profile.getContactEmail()).isEqualTo(contactEmail);
     assertThat(profile.getFamilyName()).isEqualTo(familyName);
     assertThat(profile.getGivenName()).isEqualTo(givenName);
     assertThat(profile.getDataAccessLevel()).isEqualTo(dataAccessLevel);
-    assertThat(profile.getFreeTierBillingProjectName()).isEqualTo(freeTierBillingProject);
     assertThat(profile.getEnabledInFireCloud()).isEqualTo(enabledInFirecloud);
     assertThat(profile.getContactEmailFailure()).isEqualTo(contactEmailFailure);
-    assertUser(primaryEmail, contactEmail, familyName, givenName, dataAccessLevel, firstSignInTime,
-        freeTierBillingProject);
+    assertThat(profile.getFreeTierBillingProjectName()).isNotEmpty();
+    assertUser(primaryEmail, contactEmail, familyName, givenName, dataAccessLevel, firstSignInTime);
   }
 
   private void assertUser(String primaryEmail, String contactEmail,
       String familyName, String givenName, DataAccessLevel dataAccessLevel,
-      Timestamp firstSignInTime, String freeTierBillingProject) {
+      Timestamp firstSignInTime) {
     User user = userDao.findUserByEmail(primaryEmail);
     assertThat(user).isNotNull();
     assertThat(user.getContactEmail()).isEqualTo(contactEmail);
     assertThat(user.getFamilyName()).isEqualTo(familyName);
     assertThat(user.getGivenName()).isEqualTo(givenName);
     assertThat(user.getDataAccessLevel()).isEqualTo(dataAccessLevel);
     assertThat(user.getFirstSignInTime()).isEqualTo(firstSignInTime);
-    assertThat(user.getFreeTierBillingProjectName()).isEqualTo(freeTierBillingProject);
+    assertThat(user.getFreeTierBillingProjectName()).isNotEmpty();
```

Comment:
Is this intended to enforce that a user is never created without a freeTierBillingProject?

Reply:
Kinda - I just didn't want to keep this explicit logic asserting the exact value of the billing project name. I could remove this and put it into specific tests instead, if you'd prefer. Or I could make "hasBillingProject" an explicit boolean flag here. Checking this did feel outside the scope for most of the unit tests in this file.

Reply:
I think it would be good to have at least one test that checks the exact value of the billing project name, but perfectly fine to not do it for every test.

Reply:
OK, I'm just going to drop the check from here. I added a non-empty check to one of the unit tests above, and I added an explicit value check to the retry test.